### PR TITLE
Throw explicit error if data to migrate is found and adapter option is not set

### PR DESCRIPTION
**Pull request by leplatrem**
We realized that most developers will probably upgrade without reading the release notes. 

So to avoid that, we now raise an explicit error if an old database is found and the `migrateOldData` is not explicitly set.

### CODE REVIEW

File: src/adapters/IDB.js [hunk 1]
```diff
@@ -587,7 +596,7 @@ async function migrationRequired(dbName) {
     return null;
   }
 
-  console.warn(`${dbName}: old IDB database found."`);
+  console.warn(`${dbName}: old IndexedB database found.`);
```
Comment (peterbe) [inline src/adapters/IDB.js]: IndexedDB

File: src/adapters/IDB.js [hunk 2]
```diff
@@ -238,9 +238,18 @@ export default class IDB extends BaseAdapter {
...
-      : null;
+    const hasMigrateOption = this._options.hasOwnProperty("migrateOldData");
+    const toMigrate =
+      this._options.migrateOldData || !hasMigrateOption
+        ? await migrationRequired(this.cid)
+        : null;
+
+    if (!!toMigrate && !hasMigrateOption) {
+      throw new Error(
+        "An old IndexedDB database was found, but the `migrateOldData` option was not set. " +
```
Comment (peterbe) [inline src/adapters/IDB.js]: `was not explicitly set` instead of `was not set`

File: src/adapters/IDB.js [hunk 3]
```diff
@@ -238,9 +238,18 @@ export default class IDB extends BaseAdapter {
...
+    const hasMigrateOption = this._options.hasOwnProperty("migrateOldData");
+    const dataToMigrate =
+      this._options.migrateOldData || !hasMigrateOption
+        ? await migrationRequired(this.cid)
+        : null;
+
+    if (dataToMigrate && !hasMigrateOption) {
+      throw new Error(
+        "An old IndexedDB database was found, but the `migrateOldData` option was not set. " +
+          "Check out ugprade notes https://kintojs.readthedocs.io/en/latest/upgrading/"
```
Comment (peterbe) [inline src/adapters/IDB.js]: `upgrade` not `ugprade`

File: src/adapters/IDB.js [hunk 4]
```diff
@@ -238,9 +238,19 @@ export default class IDB extends BaseAdapter {
...
+    const dataToMigrate =
+      this._options.migrateOldData || !hasMigrateOption
+        ? await migrationRequired(this.cid)
+        : null;
+
+    if (dataToMigrate && !hasMigrateOption) {
+      throw new Error(
+        "An old IndexedDB database with data was found, but the `migrateOldData` option " +
+          "was not explicitly set. " +
+          "Check out upgrade notes https://kintojs.readthedocs.io/en/latest/upgrading/"
```
Comment (peterbe) [inline src/adapters/IDB.js]: Add `#11x-to-12x` to the URL

**Comment by leplatrem** [comment]
Something is wrong. And I can't investigate it before tomorrow: https://travis-ci.org/Kinto/kinto.js/jobs/416897129

**Comment by leplatrem** [comment]
With this PR, if you don't specify advanced adapter options, we will try to open the old database. And it will create it and delete it. Each time we perform an operation (each `db.open()`)

I'm really not sure I want this. Nor force everyone to add `{ adapterOptions: { migrateOldData: false } } ` to the main constructor.

If we had a solid and efficient way (and concurrency-proof) way to check if a database exists, I would be in favor, but everything I try results to be fragile.

I would be happy to hear your suggestions :)

**Pull request closed without merging**